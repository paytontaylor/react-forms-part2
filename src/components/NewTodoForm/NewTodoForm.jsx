import { useState } from "react";

const NewTodoForm = ({ addTodo }) => {
  const [todo, setTodo] = useState("");

  const handleSubmit = e => {
    e.preventDefault();
    addTodo(todo)
    setTodo("")
  }

  const handleChange = e => {
    setTodo(e.target.value)
  }

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <label htmlFor="newTodo">New Todo</label>
        <input type="text" id="newTodo" name="newTodo" value={todo} onChange={handleChange} />
        <button>Add</button>
      </form>
    </div>
  )
}

export default NewTodoForm;