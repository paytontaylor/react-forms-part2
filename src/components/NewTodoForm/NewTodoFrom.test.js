import { render, screen } from "@testing-library/react";
import user from "@testing-library/user-event"
import NewTodoForm from "./NewTodoForm";

it("should render the NewTodoForm without crashing", () => {
  render(<NewTodoForm />)
})

it("matches snapshot", () => {
  const {  asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
})

it("should run addTodo on form submit", async () => {
  const mock = jest.fn();

  render(<NewTodoForm addTodo={mock}/>)

  const todoInput = screen.getByLabelText("New Todo");
  await user.click(todoInput);
  await user.keyboard("test todo");

  const addTodoBtn = screen.getByRole("button");
  await user.click(addTodoBtn);

  expect(mock).toHaveBeenCalled();
  expect(mock).toHaveBeenCalledWith("test todo");
})