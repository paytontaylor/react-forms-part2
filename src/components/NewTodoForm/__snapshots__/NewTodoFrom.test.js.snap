// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="newTodo"
      >
        New Todo
      </label>
      <input
        id="newTodo"
        name="newTodo"
        type="text"
        value=""
      />
      <button>
        Add
      </button>
    </form>
  </div>
</DocumentFragment>
`;
